src/ring_theory/norm.lean

```diff
@@ -130,42 +130,53 @@ end eq_prod_roots
 section eq_zero_iff
 variables [finite ι]
 
-lemma norm_eq_zero_iff_of_basis [is_domain R] [is_domain S] (b : basis ι R S) {x : S} :
-  algebra.norm R x = 0 ↔ x = 0 :=
+@[simp] lemma norm_zero [nontrivial R] [nontrivial S] [module.free R S] [module.finite R S] :
```

Comment:
[linear_map.det](https://leanprover-community.github.io/mathlib_docs/linear_algebra/determinant.html#linear_map.det) is 1 when the module is not free or not finite, to ensure it's always a monoid hom, so unfortunately we have to keep the free and finite hypotheses. `[nontrivial R]` is indeed unnecessary because the conclusion is a equality in `R`; but `nontrivial S` is necessary because the only 0x0 matrix has determinant 1 ...

Reply:
I'm wondering if it saves us some headaches redefining `linear_map.det 0 = 0` in the infinite-dimensional case. I suppose it's not going to help too much because we'd still need to specially handle the zero-dimensional case.

Reply:
It wouldn't be a monoid hom, right? In any case it doesn't matter too much.